flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -196,15 +204,21 @@ private static HiveConf createHiveConf(@Nullable String hiveConfDir) {
 		}
 
 		// create HiveConf from hadoop configuration
-		Configuration hadoopConf = HadoopUtils.getHadoopConfiguration(new org.apache.flink.configuration.Configuration());
+		Configuration hadoopConf;
 
-		// Add mapred-site.xml. We need to read configurations like compression codec.
-		for (String possibleHadoopConfPath : HadoopUtils.possibleHadoopConfPaths(new org.apache.flink.configuration.Configuration())) {
-			File mapredSite = new File(new File(possibleHadoopConfPath), "mapred-site.xml");
-			if (mapredSite.exists()) {
-				hadoopConf.addResource(new Path(mapredSite.getAbsolutePath()));
-				break;
+		if (isNullOrWhitespaceOnly(hadoopConfDir)) {
+			hadoopConf = HadoopUtils.getHadoopConfiguration(new org.apache.flink.configuration.Configuration());
```

Comment:
Let's not rely on `HadoopUtils.getHadoopConfiguration` to get the configuration. We can just call `HadoopUtils.possibleHadoopConfPaths` to get the paths and load the files by ourselves. And the loading logic should be consistent with `HiveTableUtil.getHadoopConfiguration`.